src/NationalInstruments.Analyzers/Style/ChainOfMethodsWithLambdasAnalyzer.cs

```diff
@@ -36,25 +36,53 @@ public override void Initialize(AnalysisContext context)
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
             context.RegisterSyntaxNodeAction(
-                AnalyzeSyntaxNode,
+                AnalyzeExpression,
                 SyntaxKind.ExpressionStatement,
                 SyntaxKind.EqualsValueClause,
                 SyntaxKind.Argument,
                 SyntaxKind.ArrowExpressionClause);
+
+            context.RegisterSyntaxNodeAction(
+                AnalyzeArrayInitializer,
+                SyntaxKind.ArrayInitializerExpression);
         }
 
-        private static void AnalyzeSyntaxNode(SyntaxNodeAnalysisContext context)
+        private static void AnalyzeExpression(SyntaxNodeAnalysisContext context)
         {
             var parentSyntaxNode = context.Node;
 
             // Find the invocation expression i.e
             // a method/delegate call or a property access
             // or a chain of them
             var invocationExpressionSyntax = parentSyntaxNode
-                .DescendantNodes()
+                .DescendantNodes(IsNotArrayInitializerSyntax)
                 .OfType<InvocationExpressionSyntax>()
                 .FirstOrDefault();
 
+            AnalyzeInvocationExpression(invocationExpressionSyntax, context.ReportDiagnostic);
+
+            bool IsNotArrayInitializerSyntax(SyntaxNode syntaxNode) =>
+                !syntaxNode.IsKind(SyntaxKind.ArrayInitializerExpression);
+        }
+
+        private static void AnalyzeArrayInitializer(SyntaxNodeAnalysisContext context)
+        {
+            var arrayInitializerSyntax = context.Node;
+
+            // Find only direct child invocation expressions
+            var invocationExpressionSyntaxes = arrayInitializerSyntax
+                .ChildNodes()
+                .OfType<InvocationExpressionSyntax>();
+
+            // Analyze individual invocation expressions
+            foreach (var invocationExpression in invocationExpressionSyntaxes)
```

Comment:
`invocationExpression` is a `SyntaxNode`, so I'd suffix it with "Syntax"

Reply:
Fixed [here](https://github.com/ni/csharp-styleguide/pull/65/commits/f8436c60098422a579a4049a0f9b835d1bea260a).